### PR TITLE
Platform/pm 2535/upgrade to azure messaging servicebus

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Dependency update of Microsoft.Azure.ServiceBus to Azure.Messaging.ServiceBus.

## Code changes

* **launch.json**: We build service bus subjects based on the value of an environment variable and the service that is running.
* **packages.lock.json/core.csproj**: Remove Microsoft.Azure.ServiceBus and add Azure.Messaging.ServiceBus
* **applicationCacheHostedService / InMemoryServiceBusApplicationCache**: Use the new ServiceBusClient and ServiceBusAdministrativeClient to coordinate reading/writing messages

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Services/Implementations/InMemoryServiceBusApplicationCacheService.cs [hunk 1]
```diff
@@ -19,38 +20,38 @@ public class InMemoryServiceBusApplicationCacheService : InMemoryApplicationCach
         : base(organizationRepository, providerRepository)
     {
         _subName = CoreHelpers.GetApplicationCacheServiceBusSubscriptionName(globalSettings);
-        _topicClient = new TopicClient(globalSettings.ServiceBus.ConnectionString,
-            globalSettings.ServiceBus.ApplicationCacheTopicName);
+        _serviceBusClient = new ServiceBusClient(globalSettings.ServiceBus.ConnectionString);
+        _topicMessageSender = new ServiceBusClient(globalSettings.ServiceBus.ConnectionString).CreateSender(globalSettings.ServiceBus.ApplicationCacheTopicName);
```
Comment (withinfocus) [inline src/Core/Services/Implementations/InMemoryServiceBusApplicationCacheService.cs:24]: ❓ Is it safe to call `CreateSender` in the constructor?

File: src/Core/HostedServices/ApplicationCacheHostedService.cs [hunk 2]
```diff
@@ -31,53 +35,70 @@ public class ApplicationCacheHostedService : IHostedService, IDisposable
...
         try
         {
-            await _managementClient.CreateSubscriptionAsync(new SubscriptionDescription(_topicName, _subName)
+            await _serviceBusAdministrationClient.CreateSubscriptionAsync(new CreateSubscriptionOptions(_topicName, _subName)
             {
                 DefaultMessageTimeToLive = TimeSpan.FromDays(14),
                 LockDuration = TimeSpan.FromSeconds(30),
                 EnableDeadLetteringOnFilterEvaluationExceptions = true,
-                EnableDeadLetteringOnMessageExpiration = true,
-            }, new RuleDescription("default", new SqlFilter($"sys.Label != '{_subName}'")));
```
Comment (withinfocus) [inline src/Core/HostedServices/ApplicationCacheHostedService.cs:49]: ❓ Any history on this `SqlFilter` and why it's gone now?

**Review by withinfocus — COMMENTED** [review]
I have been interested in seeing this change so this is great! Couple questions.

File: src/Core/Services/Implementations/InMemoryServiceBusApplicationCacheService.cs (hunk 1 above)
Comment (MGibson1) [reply src/Core/Services/Implementations/InMemoryServiceBusApplicationCacheService.cs:24]: Yes, that's the recommended practice: https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-performance-improvements?tabs=net-standard-sdk-2#reusing-factories-and-clients

**Review by withinfocus — APPROVED** [review]

**Review by withinfocus — APPROVED** [review]
Only lock file updates. This is gonna help a ton of PRs though!

**Pull request merged**